EKrut-Server/src/ekrut/server/managers/ServerSessionManager.java

```diff
@@ -72,7 +88,13 @@ public UserResponse logoutUser(String username) throws SQLException {
 		}
 		else {
 			connectedUsers.remove(connectedUsers.indexOf(user));
+			timer.cancel();
 			result= "OK";
+			//Remove client-user from clientUserMap
+			for(Entry<ConnectionToClient, User> entry : clientUserMap.entrySet()) {
+				if(entry.getValue().equals(user))
+					clientUserMap.remove(entry.getKey());

```

Comment:
This method should receive the `ConnectionToClient` and simply use `clientUserMap.remove(connection)`